nav2_costmap_2d/include/nav2_costmap_2d/costmap_2d.hpp

```diff
@@ -407,27 +408,36 @@ class Costmap2D
     int offset_dx = sign(dx);
     int offset_dy = sign(dy) * size_x_;
 
-    unsigned int offset = y0 * size_x_ + x0;
 
     // we need to chose how much to scale our dominant dimension,
     // based on the maximum length of the line
     double dist = std::hypot(dx, dy);
+    if (dist <= min_length) {
+      return;
+    }
+
     double scale = (dist == 0.0) ? 1.0 : std::min(1.0, max_length / dist);
 
+    x0 = (unsigned int)(x0 + dx / dist * min_length);
+    y0 = (unsigned int)(y0 + dy / dist * min_length);
+    unsigned int offset = y0 * size_x_ + x0;
+
+    unsigned int length;
     // if x is dominant
     if (abs_dx >= abs_dy) {
       int error_y = abs_dx / 2;
+      length = (unsigned int)(scale * abs_dx) - min_length;
```

Comment:
I don't 100% understand the `- min_length` both here and in the 3D voxel layer, can you explain it? I would have thought you'd want to give it the min too, I think this offsets the raytracing by the min length (e.g. if max is 10 but in is 1, then I think it should only raytrace from 0-9 rather than 1-10). I don't see how this could know what the minimum range is to know to skip over it and start there to length

Reply:
I update x0, y0 and offset in the three lines above. That makes it start from min_range. Therefore the new length needs to be reduced by min_range

Reply:
I see, thanks, can you add a comment to that effect in that calculation so that its clear to others later? Try to group those calculations together so that its not as spread out